### PR TITLE
Refactor support code

### PR DESCRIPTION
**Pull request by rycee**
Does a general cleanup of `overlay.nix`, e.g. removing unnecessary filtering on attribute values.

Replace `buildable.nix` by `ci.nix`, which contains a number of improvements over `buildable.nix`. For example, it

- honors `recurseForDerivations` to also build packages inside
  attributes,

- builds all package outputs, and

- separates buildable and cacheable packages such that packages having
  `preferLocalBuild` are not cached.

Also change the travis setup to make use of `ci.nix`.

_Note_: I have not verified that the travis changes are correct since my setup uses [gitlab ci](https://gitlab.com/rycee/nur-expressions/blob/master/.gitlab-ci.yml).

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -22,13 +22,13 @@ install:
  - travis_retry nix-channel --update
 
 script:
- - outs=$(nix-build buildable.nix) && echo Produced $outs
+ - nix-build ci.nix -A buildOutputs
```
Comment (zimbatm) [inline .travis.yml]: any way default.nix could be made the only entry-point?

Comment (rycee) [reply .travis.yml]: You mean like having a `ci` attribute in `default.nix` and do `nix-build -A ci.buildOutputs` here?

Comment (zimbatm) [reply .travis.yml]: yeah. I'm not sure about it but it's a bit tricky when there are multiple end-points. For example just today in one of my projects, I forgot to update the `shell.nix` `{ nixpkgs }` reference when I had changed it in `default.nix`

Comment (rycee) [reply .travis.yml]: It probably would work to have it in `default.nix`, you'd just need some form of self reference for the new `ci` attribute to access the packages. Or simply do `ci = import ./ci.nix { inherit pkgs }` and add "ci" as a reserved name.

But I think working on that goes outside the scope of this PR.

**Pull request merged**